gwpy/table/table.py

```diff
@@ -720,20 +720,244 @@ def filter(self, *column_filters):
+            name of the column which is used to search for clusters.
+            If index == 'omicron': use omicron clustering algortihm
+
+        window : `float`
+            window to use when clustering data points, will raise
+            ValueError if `window > 0` is not satisfied
+
+        Returns
+        -------
+        None
+
+        """
+
+        # sort table by the grequired column
+        orderidx = numpy.argsort(self[index])
+        col = self[index][orderidx]
+
+        # Find all points where the index vector changes by less than
+        # window and divide the resulting array into clusters of
+        # adjacent points
+        clusterpoints = numpy.where(numpy.diff(col) <= window)[0]
+        sublists = numpy.split(clusterpoints,
+                               numpy.where(
+                                   numpy.diff(clusterpoints) > 1)[0]+1)
+
+        # Add end-points to each cluster and find the index of the maximum
+        # point in each list
+        clustersList = [numpy.append(s, numpy.array([s[-1]+1]))
+                        for s in sublists]
+        # Find triggers that are between two clusters
+        # Happens when successive triggers do not fill the window
+        # criteria. N triggers missing --> add N clusters
+        missing_trigger = []
+        missing_trigger_idx = []
+        for i in range(len(clustersList)-1):
+            if clustersList[i+1][0] - clustersList[i][-1] > 1:
+                missing_trigger_idx.append(i+1)
+                missing_trigger.append(numpy.arange(
+                                         clustersList[i][-1]+1,
+                                         clustersList[i+1][0]))
+        # Insert them in the cluster list
+        # Need to reverse the list, otherwise the list size update
+        # is a problem
+        for i in range(len(missing_trigger))[::-1]:
+            for j in range(len(missing_trigger[i])):
+                clustersList.insert(missing_trigger_idx[i]+j,
+                                    numpy.atleast_1d(
+                                           missing_trigger[i][j]))
+
+        # Check if there are some missing points at the beginning
+        # and end of this list
+        if clustersList[0][0] != 0:
+            missing_trigger = numpy.arange(0, clustersList[0][0])
+            for i in range(len(missing_trigger)):
+                clustersList.insert(i,
+                                    numpy.atleast_1d(missing_trigger[i]))
+        if clustersList[-1][-1] != len(self):
+            missing_trigger = numpy.arange(clustersList[-1][-1]+1,
+                                           len(self))
+            for i in range(len(missing_trigger)):
+                clustersList.insert(len(clustersList)+i,
+                                    numpy.atleast_1d(missing_trigger[i]))
+
+        self.clustersList = clustersList
+
+        # keep track of the order indexes
+        self.orderidx = orderidx
+
+        return True
```

Comment:
It makes more sense to me to have these functions return the list of clusters, rather than storing them in another internal property, since the list is only ever used internal to the `identify_clusters` method.

Can you please consider that change?